hail/src/main/scala/is/hail/annotations/RegionValueBuilder.scala

```diff
@@ -444,51 +374,18 @@ class RegionValueBuilder(var region: Region) {
 
   def addRegionValue(t: PType, fromRegion: Region, fromOff: Long) {
     val toT = currentType()
-    assert(toT == t.fundamentalType)
 
     if (typestk.isEmpty) {
-      if (region.eq(fromRegion)) {
-        start = fromOff
-        advance()
-        return
-      }
-
-      allocateRoot()
+      val r = toT.copyFromType(region, t.fundamentalType, fromOff, region.ne(fromRegion))
+      start = r
```

Comment:
A change in behavior: previously addRegionValue would allocate for PPrimitive's via allocateRoot when `region != fromRegion`, now it will not, since copyFromType on PPrimitive performs no allocation when deepCopy is true.

Reply:
> since copyFromType on PPrimitive performs no allocation when deepCopy is true.

This is a bug.